### PR TITLE
Cleanup some gratuitous or inefficient uses of RootedVec

### PR DESCRIPTION
**Pull request by nox**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/6140)

### CODE REVIEW

File: components/script/dom/document.rs [hunk 1]
```diff
@@ -1028,17 +1028,11 @@ trait PrivateDocumentHelpers {
...
-            for node in NodeCast::from_ref(root.r()).traverse_preorder() {
-                let node = node.root();
-                if callback(node.r()) {
-                    nodes.push(JS::from_rooted(node.r()));
-                }
-            }
-        };
-        NodeList::new_simple_list(window.r(), &nodes)
+        let doc = self.GetDocumentElement().root();
+        let node = doc.r().map(NodeCast::from_ref);
```
Comment (Manishearth) [inline components/script/dom/document.rs]: nit: `maybe_node`, the `iter()` below is confusing without this

File: components/script/dom/domrectlist.rs [hunk 2]
```diff
@@ -19,17 +18,17 @@ pub struct DOMRectList {
...
             reflector_: Reflector::new(),
-            rects: (**rects).clone(),
+            rects: rects.map(JS::from_rooted).collect(),
             window: JS::from_rooted(window),
         }
     }
 
-    pub fn new(window: JSRef<Window>,
-               rects: &RootedVec<JS<DOMRect>>) -> Temporary<DOMRectList> {
+    pub fn new<T>(window: JSRef<Window>, rects: T) -> Temporary<DOMRectList>
```
Comment (Manishearth) [inline components/script/dom/domrectlist.rs]: I'm not sure if we should be using generics for `new`/`new_inherited`. I don't mind much, but there might be some style we're trying to maintain. @jdm @Ms2ger ?

Comment (Manishearth) [inline components/script/dom/domrectlist.rs]: Perhaps moving in a `RootedVec` would be preferable here? Then the clone is up to the caller.

**Comment by Manishearth** [comment]
Review status: all files reviewed, 3 unresolved discussions, all commit checks successful.
Reviewed files:
- components/script/dom/document.rs @ r1
- components/script/dom/domrectlist.rs @ r2
- components/script/dom/element.rs @ r2
- components/script/dom/node.rs @ r1
- components/script/dom/nodelist.rs @ r1
- components/script/script_task.rs @ r3

---

<sup>**[components/script/script_task.rs, line 1345 \[r3\]](https://reviewable.io:443/reviews/servo/servo/6140#-JqA4dYz5CKUYTUpEHm1)** ([raw file](https://github.com/servo/servo/blob/22a56e9c3f8055e15daa82757f105fb625533f42/components/script/script_task.rs#L1345)):</sup>
nit: I'd prefer some comments here explaining what is happening 

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/6140)

File: components/script/dom/domrectlist.rs (hunk 2 above)
Comment (nox) [inline components/script/dom/domrectlist.rs]: RootedVec cannot be moved.

Comment (jdm) [inline components/script/dom/domrectlist.rs]: I have no problem with using generics.

**Comment by Manishearth** [comment]
Review status: all files reviewed, 1 unresolved discussion, all commit checks successful.
Reviewed files:
- components/script/dom/document.rs @ r4
- components/script/script_task.rs @ r5

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/6140)

**Comment by Manishearth** [comment]
@bors: r+

thanks!

**Pull request merged**